acceptance/tests/validate_vendored_ruby.rb

```diff
@@ -0,0 +1,54 @@
+require 'puppet/acceptance/temp_file_utils'
+extend Puppet::Acceptance::CommandUtils
+
+test_name 'PA-1319: Validate that the vendored ruby can load gems and is configured correctly' do
+
+  step "'gem list' displays a decent list of gems" do
+    minimum_number_of_gems = 3
+    # \d+\.\d+\.\d+ describes the gem version
+    gem_re = /[\w-]+ \(\d+\.\d+\.\d+(?:, \d+\.\d+\.\d+)*\)/
+
+    agents.each do |agent|
+      gem = gem_command(agent) 
+      listed_gems = on(agent, "#{gem} list").stdout.chomp.scan(gem_re)
+      assert(listed_gems.size >= minimum_number_of_gems, "'gem list' fails to list the available gems")
+    end
+  end
+
+  step "'irb' successfully loads gems" do
+    agents.each do |agent|
+      irb = irb_command(agent)
+      ['hocon', 'deep_merge'].each do |gem|
+        stdout = on(agent, "echo \"require '#{gem}'\" | #{irb}").stdout.chomp
+        assert_match(/true/, stdout, "'irb' failed to require the #{gem} gem")
+      end
+    end
+  end
+
+  step "'gem install' successfully installs a gem with native extensions" do
+    dependencies = {
```

Comment:
It would be great if more providers and their relevant dependencies could be added here (e.g. dpkg). Better yet, we could have a hash of "provider => [<dependencies>, <gem>]" so we're not limited to sqlite. Also would be nice to have this test for Windows agents.

@ScottGarman any suggestions? I based this list off your comment on https://github.com/puppetlabs/puppet-agent/pull/1210.

Reply:
native extensions are extremely tricky on windows because they require the ruby devkit for windows. I'm not sure anyone is (or should be) using native extensions with our vendored ruby for windows

Reply:
@McdonaldSeanp does that mean users can't build nokogiri on windows? If not, do we include it in the agent? nokogiri is typically the main vendored extension gem that our users need in my experience.